applications/serial_lte_modem/src/http_c/slm_at_httpc.c

```diff
@@ -430,26 +439,48 @@ int handle_at_httpc_connect(enum at_cmd_type cmd_type)
 			if (err) {
 				return err;
 			}
-			err = at_params_unsigned_short_get(&slm_at_param_list, 3, &httpc.port);
-			if (err) {
-				return err;
+			if (at_params_unsigned_short_get(&slm_at_param_list, 3, &httpc.port)) {
+				return -EINVAL;
 			}
+			const int param_count = at_params_valid_count_get(&slm_at_param_list);
+
 			httpc.sec_tag = INVALID_SEC_TAG;
-			if (at_params_valid_count_get(&slm_at_param_list) > 4) {
-				err = at_params_unsigned_int_get(&slm_at_param_list, 4,
-							&httpc.sec_tag);
-				if (err) {
-					return err;
+			if (param_count > 4) {
+				if (at_params_unsigned_int_get(&slm_at_param_list, 4,
+							       &httpc.sec_tag)) {
+					return -EINVAL;
+				}
+			}
+			httpc.peer_verify = TLS_PEER_VERIFY_REQUIRED;
+			if (param_count > 5) {
+				if (at_params_unsigned_int_get(&slm_at_param_list, 5,
+							       &httpc.peer_verify) ||
+				    (httpc.peer_verify != TLS_PEER_VERIFY_NONE &&
+				     httpc.peer_verify != TLS_PEER_VERIFY_OPTIONAL &&
+				     httpc.peer_verify != TLS_PEER_VERIFY_REQUIRED)) {
```

Comment:
Maybe the macros could be mentioned next to the parameter values in the documentation as this is what is actually used?

Reply:
Please don't mention the macros.

When macro values change, this is the place where we need to place a wrapping layer, so that your AT interface keeps is static values.

Macro values are not supposed to be used as a static numerical values. They should be used as a symbols, and it is not considered API change, if macro value changes within the same compiled binary.
However, if they change in the binary interface, it is a binary break. And SLM's AT interface is kind of binary interface. It cannot change, if macro values change.

Reply:
If the external application using SLM were using these macros, then when compiling an hypothetical newer NCS version that has had the macro values changed, that application wouldn't have to change anything, unlike if using the hard-coded values directly as is documented in the AT command.
But it's not quite foolproof anyway. As you have said earlier it would be good at some point to make it so that these values don't change in the AT interface, even when they change internally.

Reply:
Leaving the macros out of the documentation.